workflow-templates/gradle-sonarqube.yml

```diff
@@ -181,44 +325,99 @@ jobs:
         id: aws-login-ecr
         uses: aws-actions/amazon-ecr-login@v2
 
-      - name: Test with gradle
+      - name: Create class files with gradle (single project)
+        if: matrix.project-name == 'ROOT_PROJECT'
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          arguments: classes testClasses -Pversion=${{ needs.setup.outputs.version }}
+
+      - name: Create class files with gradle (multirepo)
+        if: matrix.project-name != 'ROOT_PROJECT'
+        uses: gradle/actions/setup-gradle@v3
+        with:
+          arguments: :${{ matrix.project-name }}:classes :${{ matrix.project-name }}:testClasses -Pversion=${{ needs.setup.outputs.version }}
+
+      - name: Determine binary/source folders
+        run: |
+          echo PROJECT_BINARIES=$(find ${{ env.PROJECT_DIR }} -path '*/build/classes/java/main' -o -path '*/build/classes/kotlin/main' | sed "s/^${{ env.PROJECT_DIR }}\///" | xargs | tr ' ' ',') >> ${GITHUB_ENV}
+          echo PROJECT_SOURCES=$(find ${{ env.PROJECT_DIR }} -path '*/src/main/java' -o -path '*/src/main/kotlin' | sed "s/^${{ env.PROJECT_DIR }}\///" | xargs | tr ' ' ',') >> ${GITHUB_ENV}
+          echo PROJECT_TEST_BINARIES=$(find ${{ env.PROJECT_DIR }} -path '*/build/classes/java/test' -o -path '*/build/classes/kotlin/test' | sed "s/^${{ env.PROJECT_DIR }}\///" | xargs | tr ' ' ',') >> ${GITHUB_ENV}
+          echo PROJECT_TEST_SOURCES=$(find ${{ env.PROJECT_DIR }} -path '*/src/test/java' -o -path '*/src/test/kotlin' | sed "s/^${{ env.PROJECT_DIR }}\///" | xargs | tr ' ' ',') >> ${GITHUB_ENV}
+
+      - name: Set env 'GRADLE_SONARQUBE_TASK' (fallback to default)
+        if: env.GRADLE_SONARQUBE_TASK == ''
+        run: echo GRADLE_SONARQUBE_TASK='test sonar' >> ${GITHUB_ENV}
+
+      - name: Prepare gradle task (single project)
+        if: matrix.project-name == 'ROOT_PROJECT'
+        run: echo SONAR_TASK="${{ env.GRADLE_SONARQUBE_TASK }}" >> ${GITHUB_ENV}
+
+      - name: Prepare gradle task (multirepo)
+        if: matrix.project-name != 'ROOT_PROJECT'
+        run: |
+          IFS=' ' read -ra gradle_tasks <<< "${{ env.GRADLE_SONARQUBE_TASK }}"
+          for task in ${gradle_tasks[@]}; do
+            all_tasks+=" :${{ matrix.project-name }}:${task}"
+          done
+          echo SONAR_TASK=${all_tasks:1} >> ${GITHUB_ENV}
+
+      - name: Determine sonar final task
+        run: echo SONAR_FINAL_TASK=$(echo '${{ env.SONAR_TASK }}' | sed "s/.* //") >> ${GITHUB_ENV}
+
+      - name: Determine sonar preparation tasks
+        run: echo SONAR_PRE_TASKS=$(echo '${{ env.SONAR_TASK }}' | sed "s/\.* ${{ env.SONAR_FINAL_TASK}}//") >> ${GITHUB_ENV}
```

Comment:
any better idea instead of splitting into sonar "pre- and final-" tasks to enable a different jdk for the sonar-scanner (I actually do not like that part that much)